packages/shared/src/urbit/contact.ts

```diff
@@ -48,3 +48,106 @@ export interface ContactNews {
   who: string;
   con: Contact | null;
 }
+
+export interface ContactFieldText {
+  type: 'text';
+  value: string;
+}
+
+export interface ContactImageField {
+  type: 'look';
+  value: string;
+}
+
+export interface ContactFieldColor {
+  type: 'tint';
+  value: string;
+}
+
+export interface ContactFieldGroups {
+  type: 'set';
+  value: { type: 'flag'; value: string }[];
+}
+
+export interface ContactBookProfile {
+  nickname?: ContactFieldText;
+  bio?: ContactFieldText;
+  avatar?: ContactImageField;
+  cover?: ContactImageField;
+  color?: ContactFieldColor;
+  groups?: ContactFieldGroups;
+  status?: ContactFieldText;
+}
+
+export interface ContactBookProfileEdit {
+  nickname?: ContactFieldText | null;
+  bio?: ContactFieldText | null;
+  avatar?: ContactImageField | null;
+  cover?: ContactImageField | null;
+  color?: ContactFieldColor | null;
+  groups?: ContactFieldGroups | null;
+  status?: ContactFieldText | null;
+}
+
+// first element is the contact's profile, second is any user overrides
+export type ContactBookEntry = [ContactBookProfile, ContactBookProfile];
+
+export type ContactsAllScryResult1 = Record<string, ContactBookProfile>;
```

Comment:
What's up with the `1` here? Seems unnecessary?

Reply:
Trying to be better about explicitly versioning scry results